pkg/client/auditor/auditor.go

```diff
@@ -100,16 +100,22 @@ func DefaultAuditor(
 
 func (a *defaultAuditor) Run(
 	interval time.Duration,
+	singleRun bool,
 	stopc <-chan struct{},
 	donec chan<- struct{},
-) {
+) (err error) {
 	defer func() { donec <- struct{}{} }()
 	a.logger.Infof("starting auditor with a %s interval ...", interval)
-	err := repeat(interval, stopc, a.audit)
-	if err != nil {
-		return
+	if singleRun {
```

Comment:
I suppose this could've been easily achieved without any code change, by calling the `defaultAuditor.audit` method directly in the unit test (since unit tests test the implementation, anyway, not the interface). Not a big deal, though.

Reply:
Agree, but I liked to test Run method